prow/inrepoconfig/api/api_test.go

```diff
@@ -0,0 +1,378 @@
+	tests := []struct {
+		name         string
+		inrepoconfig *InRepoConfig
+		configModify func(*config.Config)
+		verifyFunc   func(config.Presubmit) error
+		errExpected  bool
+	}{
+		{
+			name: "Verify DefaultPresubmitFields is executed",
+			inrepoconfig: &InRepoConfig{Presubmits: []config.Presubmit{{
+				JobBase: config.JobBase{
+					Name: "test-job",
+				},
+			}}},
+			verifyFunc: func(ps config.Presubmit) error {
+				if ps.Context != "test-job" {
+					return fmt.Errorf(`Expected ps.Context to be "test-job", was %q`, ps.Context)
+				}
+				if ps.Agent != "kubernetes" {
+					return fmt.Errorf(`Expected agent to be "kubernetes", was %q`, ps.Agent)
+				}
+				return nil
+			},
+		},
+		{
+			name: "Verify decoration config gets set",
+			inrepoconfig: &InRepoConfig{Presubmits: []config.Presubmit{{
+				JobBase: config.JobBase{
+					Name:          "test-job",
+					UtilityConfig: config.UtilityConfig{Decorate: true},
+				},
+			}}},
+			configModify: func(c *config.Config) {
+				c.Plank.DefaultDecorationConfig = &prowapi.DecorationConfig{
+					CookiefileSecret: "my-secret",
+					UtilityImages: &prowapi.UtilityImages{
+						CloneRefs:  "a",
+						InitUpload: "b",
+						Entrypoint: "c",
+						Sidecar:    "d",
+					},
+					GCSConfiguration: &prowapi.GCSConfiguration{
+						PathStrategy: "explicit",
+					},
+					GCSCredentialsSecret: "e",
+				}
+			},
+			verifyFunc: func(ps config.Presubmit) error {
+				if ps.DecorationConfig == nil || ps.DecorationConfig.CookiefileSecret == "" {
+					return errors.New("Expected decoration config to get set")
+				}
+				return nil
+			},
+		},
+		{
+			name: "Verify Presets get set",
+			inrepoconfig: &InRepoConfig{Presubmits: []config.Presubmit{{
+				JobBase: config.JobBase{
+					Name:   "test-job",
+					Labels: map[string]string{"my-ps": "true"},
+				},
+			}}},
+			configModify: func(c *config.Config) {
+				c.Presets = []config.Preset{
+					{
+						Labels: map[string]string{"my-ps": "true"},
+						Env:    []corev1.EnvVar{{Name: "my-env-var"}},
+					},
+				}
+			},
+			verifyFunc: func(ps config.Presubmit) error {
+				for _, container := range ps.Spec.Containers {
+					for _, envVar := range container.Env {
+						if envVar.Name == "my-env-var" {
+							return nil
+						}
+					}
+				}
+				return errors.New("Did not find env var on container")
+			},
+		},
+		{
+			name: "Verify checkconfigapis ValidatePresubmitJob is executed",
+			inrepoconfig: &InRepoConfig{Presubmits: []config.Presubmit{{
+				JobBase: config.JobBase{
+					Name: "Very long job name because short ones can be done by everyone!!!",
+				},
+			}}},
+			errExpected: true,
+		},
+		{
+			name: "Verify regexes get set",
+			inrepoconfig: &InRepoConfig{Presubmits: []config.Presubmit{{
+				JobBase: config.JobBase{
+					Name: "test-job",
+				},
+			}}},
+			verifyFunc: func(ps config.Presubmit) error {
+				// The config.Presubmit.re field is unexported, so we have to create
```

Comment:
I vaguely remember having a helper to remove regexes for this use?

Reply:
This test is specifically about verifying that they are there, so removing them won't really help with that :)

Reply:
Ah, of course